.github/workflows/build-test-ship.yaml

```diff
@@ -357,6 +364,15 @@ jobs:
             fi
           done
 
+      - name: Launch elasticsearch via docker-compose
+        if: ${{ inputs.uses_elasticsearch }}
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        shell: bash
+        run: |
+          docker-compose pull --include-deps elasticsearch
+          docker compose up -d --wait elasticsearch
```

Comment:
At one point I floated the idea of just adding `elasticsearch` to the `depends_on` list via [`docker-compose.ci-override.yml`](https://github.com/nicheinc/development/pull/312) and then letting the usual `docker-compose` dependency management handle spinning it up. Is there a reason to do so in a separate manual step instead?

Reply:
In [this commit](https://github.com/nicheinc/actions/pull/15/commits/4ba77e526346b2be3578f5153ea35b4b3fc86608), I tried that approach. Here is the [`docker-compose.ci-override.yml` with `depends_on: elasticsearch`](https://github.com/nicheinc/development/pull/312/commits/238c69e06c545a4639214659f6361e41f72d3ff9). I can't find the `search` workflow run log, but this run failed because elasticsearch wasn't ready.

Then in [this commit](https://github.com/nicheinc/actions/pull/15/commits/2f35333d013d3bbcf0d1cbc113e1b5d3bd7670a1), I tried to add the `--wait` flag so that docker compose would wait for the dependencies to be healthy, but that also failed.

Now that I think about it, perhaps those failures were because there was no index. I'll go back and try out adding `depends_on: elasticsearch` with new indexer changes and see if that works. If not, I think this is the best option for the reasons above.

Reply:
Following up on this. I changed the following:
1. [Added `elasticsearch` to `docker-compose.ci-override.yml`](https://github.com/nicheinc/development/pull/312/commits/a9a1c29d491448a8985199489f7a341a5025bcab)
2. [Removed the `uses_elasticsearch` input and elasticsearch step in the workflow](https://github.com/nicheinc/actions/pull/15/commits/dd3a098bafddd9df79e720fa6f8e6ae7b30b663b)
3. [Remove the `uses_elasticsearch` input in search's `ci.yaml`.](https://github.com/nicheinc/search/pull/238/commits/2f4de31b3a88999d53a4d00ac729e350d56c821c)

The workflow [failed in the subsequent run.](https://github.com/nicheinc/search/actions/runs/3423514460/jobs/5702210307#step:17:35) To verify that our changes really did come into effect for this run, we can look at [our workflow's environment's docker-compose.override file](https://github.com/nicheinc/search/actions/runs/3423514460/jobs/5702210307#step:11:31) and see the changes.

I agree that adding a new input and having an optional step just for elasticsearch is not ideal. Do you have an idea for a more elegant solution?

Reply:
Thanks so much for trying that out! Recapping [this thread](https://nicheinc.slack.com/archives/C026TLM8HSL/p1667946063329399), I think this is definitely just a timing issue. I worry we're going to run into it again later with other services that can start up slowly due to external connections. 🤔

But for now, I think we should go with your plan A to start up `elasticsearch` first, though I would like to suggest a sort of middle ground. Rather than taking an optional `uses_elasticsearch` workflow input, what do you think about this approach?

1. Keep the `depends_on` override in `docker-compose.ci-override.yml`
2. In the [`parse-deps`](https://github.com/nicheinc/actions/blob/621432232d65262387320a9d5a28588f423b940f/.github/workflows/build-test-ship.yaml#L297-L298) step, use `deps` to set a `uses-elasticsearch` step output (based on whether `deps` contains `elasticsearch`)
3. Use `parse-deps.uses-elasticsearch` as the guard for the `Launch elasticsearch via docker-compose` step

The benefit would be that if we can figure out how to remove the `Launch elasticsearch` step in the future, we wouldn't have to update all the service `ci.yaml`s.

Reply:
I did some more testing into this yesterday and I have good news. The issue and my proposed solution are much simpler than I had thought: `elasticsearch` just takes more time to start than every other tool/service. The previous proposed changes with elasticsearch starting within its own step fixed this issue by giving elasticsearch more time to get ready before the service tries to connect. However, this extra time padding does not have to be _before_ the "start serivces" step, it can also be _after_.

As such, I'm proposing a much[ simpler band-aid fix where we just sleep for some amount of time](https://github.com/nicheinc/actions/pull/17/files#diff-9eacf7f2810cbe72e6e439ad8b0ffb68c86195b3f2985cededf05b73e2efd627R368) (1 minute has been reliable for me) after we call `docker-compose up` to allow the services some time to make a healthy connection with elasticsearch. Sleeping for an arbitrary amount of time is also not ideal, but I think this band-aid is better since it may help with other dependencies with long startups down the line, as well as being easier to adjust/replace.

Additionally, note on a long-term plan of using `--wait`: I believe the reason for why `--wait` stalls is because it needs every dependency to be healthy before it continues. However, as we know from the [CI Service Status spreadsheet](https://docs.google.com/spreadsheets/d/18JOH27RaSS3KXUxSqEmTYZzPjYRig-DNTTJ1s4kqzrc/edit), tons of services are still failing on start up. Additionally, if we switched to `--wait` once every service was working in CI, then if just one service had a CI-breaking change, then any service that depended upon it would stall during the "start services" step.

Reply:
That's great news! Kudos for that investigation and for your proposed short- and long-term fixes!

Reply:
Thanks! I created an [Asana task](https://app.asana.com/0/1202563967140527/1203332769997382/f) to eventually revisit this solution. I'm going to update this PR's title/description and merge. Sound good?

Reply:
Sounds good, but with regard to this...

> This PR should be updated to pin to the newest `search` commit before being merged.

...what are you thinking for the pinned commit? It's currently pinned to [`b227b6`](https://github.com/nicheinc/search/pull/238/commits/b227b65be52e79cb9f071e751f6a78b51768e1ed), which is part of https://github.com/nicheinc/search/pull/238, which hasn't been QA'd or merged yet. Do we actually need to update the `search` submodule in that PR, or can we leave it where it is for now?